### PR TITLE
Post endpoint to create rest stop

### PR DESCRIPTION
**Pull request by connie-code**
**Related Issues**: [https://caravanning.atlassian.net/browse/CA-92](https://caravanning.atlassian.net/browse/CA-92)
[https://caravanning.atlassian.net/browse/CA-96](https://caravanning.atlassian.net/browse/CA-96) - Here I added the Long and Lat of the requested location to the rest stop table. Also, added the Long + Lat to the trip table for the start and end location. 

**Description**: Post/restStop
Create POST request so user can request a rest stop to to trip they are invited to

**Testing PR**: 
You can test this by following these instructions:
1. Run the server ``` node server.js ```
2. Open or Download [Postman](https://www.postman.com/downloads/).
3. To test the functionality create a ``` POST ``` endpoint ``` localhost:8080/restStop ```
image
4. Before clicking ``` Send ```, navigate to the ``` Body ``` tab in Postman. This is where you can specify the trip information of the trip you would like to create. Next, click on the radio button ``` raw ``` and select ``` JSON ``` from the dropdown (located on the far right of the radio buttons). Then, in the text editor provided below, insert the trip information in JSON form. Be sure to include: ``` trip_id ```, ``` req_location ```, ``` loc_long ```, ``` loc_lat ```. 
image
5. Once the above steps are complete, you may click ``` Send ``` (the blue button located towards the top of the application) and you will see the word ``` Created ``` on the bottom when you scroll down. 
image

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -100,6 +100,22 @@ app.post('/invitations', (req, res) => {
...
 });
 
+app.post('/restStop', (req, res) => {
+	const { trip_id, req_location, loc_long, loc_lat } = req.body;
+	pool.query(
+		'INSERT INTO reststop(tripid, location, loc_long, loc_lat) VALUES ($1, $2, $3, $4)',
+		[trip_id, req_location, loc_long, loc_lat],
+		(err, results) => {
+			if (err) {
+				console.log('Error when inserting invitation', err);
```
Comment (Quetourah) [inline server.js]: this error message should be updated to return proper information

Comment (connie-code) [reply server.js]: Thank you. I will fix that.

File: server.js [hunk 2]
```diff
@@ -100,6 +100,22 @@ app.post('/invitations', (req, res) => {
 	);
 });
 
+app.post('/restStop', (req, res) => {
+	const { trip_id, req_location, loc_long, loc_lat } = req.body;
```
Comment (kcode20) [inline server.js]: Why are we naming this `req_location` instead of just `location`? I think it is more intuitive to just name it `location`.

**Comment by kcode20** [comment]
Don't forget to add it to the [documentation](https://docs.google.com/spreadsheets/d/1UZ71wKQwlh75fb2J0UOv1ezPOFAHq4x4PAbpWzAU9Ww/edit?usp=sharing)

File: server.js [hunk 3]
```diff
@@ -100,6 +100,22 @@ app.post('/invitations', (req, res) => {
 	);
 });
 
+app.post('/restStop', (req, res) => {
```
Comment (kcode20) [inline server.js]: can we also make it `reststop`? It's industry standard to keep all endpoints the name of the table and all lowercase.

**Pull request merged**